### PR TITLE
[RFC] add VCard::getByType()

### PR DESCRIPTION
**Pull request by kbond**
Hello! I had a requirement for this feature so I thought I would create a PR to discuss.

### CODE REVIEW

File: lib/Component/VCard.php [hunk 1]
```diff
@@ -411,6 +411,24 @@ function preferred($propertyName) {
     }
 
     /**
+     * Returns a specific type of multi fields (ADR, TEL, or EMAIL).
```
Comment (evert) [inline lib/Component/VCard.php]: Perhaps:

```
Returns a property with a specific TYPE value (ADR, TEL, or EMAIL)

This function will return null if the property does not exist. If there are multiple properties with
the same TYPE value, only one will be returned.
```

File: lib/Component/VCard.php [hunk 2]
```diff
@@ -411,6 +411,24 @@ function preferred($propertyName) {
...
+     *
+     * @param string $propertyName
+     * @param string $type
+     *
+     * @return VObject\Property|null
+     */
+    function getForType($propertyName, $type) {
```
Comment (evert) [inline lib/Component/VCard.php]: I would suggest renaming this to `getByType` to be more in line with other functions.

File: lib/Component/VCard.php [hunk 3]
```diff
@@ -411,6 +411,24 @@ function preferred($propertyName) {
...
+        foreach ($this->select($propertyName) as $field) {
+            if (isset($field['TYPE']) && $field['TYPE']->getValue() === $type) {
```
Comment (evert) [inline lib/Component/VCard.php]: What happens in a case when there's a property with multiple TYPE parameters, or a multi-value TYPE parameter? I think Parameter objects have a `->has` function that should be used instead.

File: lib/Component/VCard.php [hunk 4]
```diff
@@ -411,6 +411,24 @@ function preferred($propertyName) {
...
+                return $field;
+            }
+        }
+
+        return null;
```
Comment (evert) [inline lib/Component/VCard.php]: This line is not needed, as by default it will return `null`.

File: lib/Component/VCard.php (hunk 2 above)
Comment (kbond) [inline lib/Component/VCard.php]: Sure, what do you suggest?

File: lib/Component/VCard.php (hunk 3 above)
Comment (kbond) [inline lib/Component/VCard.php]: something like:

```php
if (isset($field['TYPE']) && $field['TYPE']->has($type)) {
    // ...
}
```

**Comment by kbond** [comment]
Ok, I have made the suggested changes.

**Pull request merged**